File: 1827. Minimum Operations to Make the Array Increasing.py
#PYTHON
#* 1827. Minimum Operations to Make the Array Increasing
nums = [1,5,2,4,1]
from typing import List
#* MY SOL
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        operations = 0
        for i in range(1, len(nums)):
            if nums[i] <= nums[i - 1]:
                operations += nums[i - 1] - nums[i] + 1
                nums[i] = nums[i - 1] + 1
        return operations


print(Solution().minOperations(nums))

File: 1827. Minimum Operations to Make the Array Increasing.js
// 1827. Minimum Operations to Make the Array Increasing
nums = [1,5,2,4,1]
/**
 * @param {number[]} nums
 * @return {number}
 */
var minOperations = function(nums) {
    let operations = 0;
    for (let i = 1; i < nums.length; i++) {
        if (nums[i] <= nums[i - 1]) {
            operations += nums[i - 1] - nums[i] + 1;
            nums[i] = nums[i - 1] + 1;
        }
    }
    return operations;
}
console.log(minOperations(nums))
